catalog/core/catalog-core-api/src/main/java/ddf/catalog/validation/MessageValidator.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.validation;
+
+import ddf.catalog.operation.ProcessingDetails;
+import java.util.Set;
+
+public interface MessageValidator {
+
+  /**
+   * Get the identifier of this validator
+   *
+   * @return the identifier, or the empty {@link String} when no identifier exists
+   */
+  String getId();
+
+  /**
+   * Determine whether this {@link MessageValidator} has the capacity to validate the message(s) of
```

Comment:
✏️ It seems like we should be using the term `warning` instead of `message` since that is the term used by `SourceProcessingDetails`.